plugins/system/pressure

```diff
@@ -0,0 +1,331 @@
+
+=head1 CONFIGURATION
+
+Simply create a symlink in your plugins directory like with any other plugin.
+No additional configuration needed, no specific user required (typically).
+
+If you want to configure alerts, just add "warn_" or "crit_" in front of the
+internal name.
+
+Optional configuration examples:
+
+[pressure]
+env.resources cpu io memory       - Specify the resources to monitor. Leave one
+                                    out if you don't want this one to be
+                                    monitored.
+env.intervals avg10 avg60 avg300  - Sepcify the average intervals to monitor.
+                                    Leave one out if you don't want this one to
+                                    be monitored
+env.scopes some full              - Specify the scopes to monitor. Leave one out
+                                    If you don't want it to be monitored.
+env.summary_interval avg300       - Specify the interval to be used for the
+                                    summary-graph.
+env.warn_psi_cpu_avg300_some 5    - Set a warning-level of 5 for
+                                    "psi_cpu_avg300_some"
+env.crit_psi_io_total_full 2000   - Set a critical-level of 2000 for
+                                    "psi_io_total_full"
+
+=head1 AUTHOR
+
+2022, HaseHarald
+
+=head1 LICENSE
+
+LGPLv3
+
+=head1 BUGS
+
+=head1 TODO
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf
+
+=cut
+
+
+# This file contains a munin-plugin to graph the psi (pressure) for CPU, Memory
+# and IO, as reported by the Linux kernel.
+#
+# This is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this plugin.  If not, see <http://www.gnu.org/licenses/>.
+
+
+resource_defaults=('cpu' 'io' 'memory')
+interval_defaults=('avg10' 'avg60' 'avg300')
+scope_defaults=('some' 'full')
+pressure_dir=${pressure_dir:-'/proc/pressure/'}
+pressure_resources=${resources[@]:-${resource_defaults[@]}}
+pressure_intervals=${intervals[@]:-${interval_defaults[@]}}
+pressure_scopes=${scopes[@]:-${scope_defaults[@]}}
+summary_interval=${summary_interval:-avg300}
+
+check_autoconf() {
+  if [ -d "${pressure_dir}" ]; then
+    printf "yes\n"
+  else
+    printf "no (%s not found)\n" ${pressure_dir}
+  fi
+}
+
+get_pressure_value() {
+  resource=$1
+  interval=$2
+  scope=${3:-some}
+  grep "$scope" ${pressure_dir}//${resource} | grep -o -E "${interval}=[0-9]{1,}(\.[0-9]{1,}){0,1}" | cut -d '=' -f 2
```

Comment:
Just one slash is sufficient for the `grep` path, oder?

Please quote the path.

Regarding `{1,}` in the regular expression: wouldn't be `+` an easier readable way of expressing this?

Reply:
Well, that depends on your definition of "easy". The way it currently is set, the regex works the same way throughout. It's all the same pattern. Whereas your solution (using `+`) brings in a different notation, where one reading this will need to think about it again.

Off course one could argue that this is still common regex that most everybody knows by heart, if they know regex at all, and that regular changes can help to keep you awake. But still its opinion either way around.

The double slash is a typo by the way. I'm going to fix that.

Reply:
OK, you thought about the possible ways of expressing this. So I am happy with your choice.